### PR TITLE
Introduce production-like matrix workflow for Phase 2 performance profiling

### PR DESCRIPTION
**Pull request by kpedro88**
As discussed a few days ago, this type of production-like workflow will be used in the future for Phase 2 performance profiling as an (optional) part of PR tests. @slava77 @fabiocos 

Details:
1. Remove DQM and validation steps from RECO part of workflow.
2. Put PAT step into separate miniAOD part of workflow.
3. HARVEST part of workflow is removed (condition added to `MatrixReader` to skip steps that are `None`).
4. Set 2023 PU workflows to use 200PU by default.
5. Workflows are forwarded to the standard matrix for D35 ttbar, 0PU and 200PU (classic mixing): 27434.21, 27634.21

Setting this up for premixing might be more difficult... @makortel

### CODE REVIEW

**Comment by makortel** [comment]
> Setting this up for premixing might be more difficult... 

@kpedro88 We should probably discuss about that.

**Comment by kpedro88** [comment]
@makortel sure, I opened the PR already in order to get any comments.

**Comment by kpedro88** [comment]
please test workflow 27434.21

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2916,11 +2916,12 @@ def gen2018HiMix(fragment,howMuch):
 for ds in defaultDataSets:
     key='MinBias_14TeV_pythia8_TuneCUETP8M1'+'_'+ds
     name=baseDataSetReleaseBetter[key]
-#    if '2017' in name or '2018' in name:
     if '2017' in name:
     	PUDataSets[ds]={'-n':10,'--pileup':'AVE_35_BX_25ns','--pileup_input':'das:/RelValMinBias_13/%s/GEN-SIM'%(name,)}
     elif '2018' in name:
     	PUDataSets[ds]={'-n':10,'--pileup':'AVE_50_BX_25ns','--pileup_input':'das:/RelValMinBias_13/%s/GEN-SIM'%(name,)}
+    elif '2023' in name:
+    	PUDataSets[ds]={'-n':10,'--pileup':'AVE_200_BX_25ns','--pileup_input':'das:/RelValMinBias_13/%s/GEN-SIM'%(name,)}
```
Comment (makortel) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: Should 2023 use `RelValMinBias_14TeV`?

**Comment by kpedro88** [comment]
please test workflow 27434.21

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
@prebello @zhenhu any comment/objection?

**Comment by zhenhu** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Comment by smuzaffar** [comment]
@kpedro88 , workflow 27634.21 input is not found. Is /RelValMinBias_14TeV/1/GEN-SIM here is correct? 
```
the query is file dataset = /RelValMinBias_14TeV/1/GEN-SIM
DAS succeeded after 1 attempts 0
found files:  []
```

**Comment by kpedro88** [comment]
@smuzaffar no, the default dataset needs to be updated. I can submit a PR to fix this, but maybe I should remove this workflow from the regular matrix as well? We probably don't need to run 200PU in IBs...

**Comment by kpedro88** [comment]
see #26046

**Pull request merged**